src/MPQEditor/MPQEditor.ts

```diff
@@ -0,0 +1,647 @@
+
+  updateSectionOfLayer(name: string, section: string, value: string) {
+    let layer = this._content[MPQData._layersKey].find(
+      (x: any) => x["name"] === name
+    );
+    if (layer) {
+      layer[section] = value;
+    }
+  }
+
+  getSelected(): Set<string> {
+    return this._selectedNodes;
+  }
+
+  clearSelection() {
+    this._selectedNodes.clear();
+  }
+
+  removeModelNodesFromLayers(names: any) {
+    names.forEach((name: any) => {
+      let foundIndex = this._content[MPQData._layersKey].findIndex(
+        (x: any) => x["name"] === name
+      );
+      if (foundIndex > -1) {
+        this._content[MPQData._layersKey].splice(foundIndex, 1);
+      }
+    });
+    this.filterModelNodesbyContent();
+  }
+
+  getVisqPath(): string {
+    return this._visqPath;
+  }
+
+  setVisqPath(path: string): void {
+    this._visqPath = path;
+  }
+}
+
+export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
+  public static readonly viewType = "one.editor.mpq";
+  public static readonly fileExtension = ".mpq.json";
+
+  private _disposables: vscode.Disposable[] = [];
+  private _mpqDataMap: any = {};
+
+  public static register(context: vscode.ExtensionContext): void {
+    const provider = new MPQEditorProvider(context);
+    const registrations = [
+      vscode.window.registerCustomEditorProvider(
+        MPQEditorProvider.viewType,
+        provider,
+        {
+          webviewOptions: { retainContextWhenHidden: true },
+        }
+      ),
+      // Add command registration here
+      vscode.commands.registerCommand(
+        "one.editor.mpq.showFromDefaultExplorer",
+        (uri) => {
+          MPQEditorProvider.createMPQJson(uri);
+        }
+      ),
+      vscode.commands.registerCommand(
+        "one.editor.mpq.showFromOneExplorer",
+        (uri) => {
+          MPQEditorProvider.createMPQJson(uri);
+        }
+      ),
+    ];
+
+    registrations.forEach((disposable) =>
+      context.subscriptions.push(disposable)
+    );
+  }
+
+  public static createMPQJson(uri: vscode.Uri): void {
+    const extName = path.parse(uri.path).ext.slice(1);
+    if (extName !== "circle") {
+      return;
+    }
+
+    const dirPath = path.parse(uri.path).dir;
+    const modelName = path.parse(uri.path).name;
+
+    const circleName = path.parse(uri.path).base;
+
+    const content =
+      '{"default_quantization_dtype": "uint8",' +
+      '"default_granularity": "channel",' +
+      '"layers": [],' +
+      '"model_path": ' +
+      '"' +
+      circleName +
+      '"' +
+      "}";
+
+    const findInputPath = (mpqName: string): string => {
+      const maxIters = 5;
+      for (let i = 0; i < maxIters; i++) {
+        const mpqPath: string = path.join(
+          dirPath,
+          mpqName + MPQEditorProvider.fileExtension
+        );
+        if (!fs.existsSync(mpqPath)) {
+          return mpqName + MPQEditorProvider.fileExtension;
+        }
+        mpqName = mpqName + "(1)";
+      }
+      return "";
+    };
```

Comment:
This part seems to be making 'Filename.mpq.json(1).mpq.json(1).mpq....' kinds, which is not expected.

Reply:
@dayo09 
Thank you for your review.
Actually `mpqName` doesn't contain extension part, so for the `circle` file named `model.circle` it will produce
1. `model.mpq.json`
2. `model(1).mpq.json`
3. `model(1)(1).mpq.json`
4. `model(1)(1)(1).mpq.json`
5. `model(1)(1)(1)(1).mpq.json`
then it will ask the user to resolve the `json` name.
But another naming scheme can be introduced as well.

Reply:
@stamalakhov , it seems you don't test your own code.
From now on, I want to request a test code for each change or a test report or screen capture of UI or log. Reviewers are not testers. Please don't make reviewers test your code.

Reply:
@seanshpark 
Thank you for your review.
Sorry. I've tested that in a draft, so it produced the sequence mentioned above https://github.com/Samsung/ONE-vscode/pull/1510#discussion_r1141725449.
I've tested this PR, that it can be successfully compiled and that the part is not changed from draft.